### ISSUE TITLE
certificate.Export(X509ContentType.Pfx) does not export EC private key

### ISSUE THREAD

**Issue by shaan1337** [dotnet/runtime#37735]
### Description
Certificates with an EC public/private key are not exported properly.
In the snippet below, certificate.Export(X509ContentType.Pfx) or certificate.Export(X509ContentType.Pkcs12) both do not export the private key. The same code snippet works if you change the key to an RSA key instead.

```
            //generate a certificate with EC public/private key pair
            var ecdsa = ECDsa.Create();
            ecdsa.GenerateKey(ECCurve.NamedCurves.nistP256);
            var dname = new X500DistinguishedName("CN=test");
            var certificateRequest = new CertificateRequest(dname, ecdsa,  HashAlgorithmName.SHA256);
            var certificate = certificateRequest.CreateSelfSigned(DateTimeOffset.UtcNow.AddMonths(-1), DateTimeOffset.UtcNow.AddMonths(1));
            Console.WriteLine(certificate.HasPrivateKey); //prints True

            //export the first certificate to PKCS12 format then create a new certificate from it
            var certificate2 = new X509Certificate2(certificate.Export(X509ContentType.Pfx));
            Console.WriteLine(certificate2.HasPrivateKey); //prints False
```
### Configuration
* Which version of .NET is the code running on?
.NET core 3.1.201

* What OS and version, and what distro if applicable?
Ubuntu 16.04

* What is the architecture (x64, x86, ARM, ARM64)?
x64

* Do you know whether it is specific to that configuration?
No

**Comment by Lougarou** [comment]
seems to be working on MacOS High Sierra 10.13.6 and dotnet 3.1.100

**Comment by bartonjs** [comment]
I think this is a dupe of #27130.  Basically, there's some bug in OpenSSL 1.0.2, fixed in OpenSSL 1.1.x, where it fails to match the key to the cert when loading the PFX.

This is fixed in master, because we now use a custom PFX loader.  On 3.1 the easiest fixes are: 

* Upgrade to Ubuntu 18.04
* Obtain OpenSSL 1.1.1 and run your dotnet process with the `CLR_OPENSSL_VERSION_OVERRIDE` environment variable set to `1.1`
* Use the Pkcs12Info class and write a custom PFX loader.  (Not very easy).

**Comment by shaan1337** [comment]
@bartonjs I can confirm it's the same bug, openssl 1.1.1 fixes it, thanks!
